app/controllers/review_item/accept.js

```diff
@@ -148,27 +211,36 @@ export default Ember.Controller.extend({
         promises.push(existing[id].destroyRecord());
       }
 
-      Ember.RSVP.all(promises)
-        .then(() => {
-          // save item
-          var item = this.get("item");
-          item.set("packageType", this.get("itemType")); // this throws error in onItemTypeChange so using itemSaving as workaround
-          item.set("donorDescription", this.get("reviewItem.formData.donorDescription"));
-          item.set("donorConditionId", this.get("reviewItem.formData.donorConditionId"));
-          if (this.get("isAccepting")) {
-            item.set("state_event", "accept");
-          } else if (item.get("isDrafted")) {
-            item.set("state_event", "submit");
-          } else  {
-            item.set("state_event", null);
-          }
-          item.save().finally(() => {
-            this.set("itemSaving", false);
-            loadingView.destroy();
-            this.transitionToRoute("review_offer.items");
-            this.get("reviewOfferController").set("displayCompleteReviewPopup", this.get("offer.allItemsReviewed") && this.get("offer.isUnderReview"));
-          });
+      Ember.RSVP.all(promises).then(() => {
+        // save item
+        var item = this.get("item");
+        item.set("packageType", this.get("itemType")); // this throws error in onItemTypeChange so using itemSaving as workaround
+        item.set(
+          "donorDescription",
+          this.get("reviewItem.formData.donorDescription")
+        );
+        item.set(
+          "donorConditionId",
+          this.get("reviewItem.formData.donorConditionId")
+        );
+        if (this.get("isAccepting")) {
+          item.set("state_event", "accept");
+        } else if (item.get("isDrafted")) {
+          item.set("state_event", "submit");
+        } else {
+          item.set("state_event", null);
+        }
+        item.save().finally(() => {
```

Comment:
you need add a return statement here. Otherwise the promise will resolve before the save action is actually completed, which could result in asynchronous drama